packages/ember-glimmer/lib/ember-metal-views/index.js

```diff
@@ -13,12 +15,41 @@ class DynamicScope {
   }
 }
 
+function maybeUpdate(lastTagValue, renderedRoots) {
+  if (CURRENT_TAG.validate(lastTagValue)) { return; }
+  for (let i = 0; i < renderedRoots.length; ++i) {
+    let view = renderedRoots[i];
+    view.renderer.rerender(view);
+  }
+}
+
+function registerView(view, renderedRoots, scheduler) {
+  if (!renderedRoots.length) {
+    run.backburner.on('begin', scheduler);
+  }
+  renderedRoots.push(view);
+}
+
+function deregisterView(view, renderedRoots, scheduler) {
+  let viewIndex = renderedRoots.indexOf(view);
+  if (~viewIndex) {
+    renderedRoots.splice(viewIndex, 1);
+    if (!renderedRoots.length) {
+      run.backburner.off('begin', scheduler);
+    }
+  }
+}
+
 class Renderer {
   constructor({ dom, env, destinedForDOM = false }) {
     this._root = null;
     this._dom = dom;
     this._env = env;
     this._destinedForDOM = destinedForDOM;
+    this._renderedRoots = [];
+    this._scheduler = function scheduleMaybeUpdate() {
+      run.backburner.schedule('render', null, maybeUpdate, CURRENT_TAG.value(), this._renderedRoots);
+    }.bind(this);
```

Comment:
Something like how you can pass a context to schedule? That would be nice.
On May 5, 2016 1:37 PM, "Stefan Penner" <notifications@github.com> wrote:

> In packages/ember-glimmer/lib/ember-metal-views/index.js
> <https://github.com/emberjs/ember.js/pull/13430#discussion_r62250689>:
>
> >  class Renderer {
> >    constructor({ dom, env, destinedForDOM = false }) {
> >      this._root = null;
> >      this._dom = dom;
> >      this._env = env;
> >      this._destinedForDOM = destinedForDOM;
> > +    this._renderedRoots = [];
> > +    this._scheduler = function scheduleMaybeUpdate() {
> > +      run.backburner.schedule('render', null, maybeUpdate, CURRENT_TAG.value(), this._renderedRoots);
> > +    }.bind(this);
>
> sorry, i was mostly thinking outloud, I think it would be great if someday
> we merely have some nice protocol to avoid that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/emberjs/ember.js/pull/13430/files/09bbce3dbdaebc6842046e3a4e7c88742d948f49#r62250689>
>